rocAL/rocAL/source/pipeline/master_graph.cpp

```diff
@@ -297,33 +297,34 @@ MasterGraph::build()
     start_processing();
     return Status::OK;
 }
-Image *
-MasterGraph::create_loader_output_image(const ImageInfo &info)
-{
+
+rocalTensor *
+MasterGraph::create_loader_output_tensor(const rocalTensorInfo &info) {
     /*
-    *   NOTE: Output image for a source node needs to be created as a regular (non-virtual) image
+    *   NOTE: Output tensor for a source node needs to be created as a regular (non-virtual) tensor
     */
-    auto output = new Image(info);
-
+    auto output = new rocalTensor(info);
     if(output->create_from_handle(_context) != 0)
-        THROW("Creating output image for loader failed");
+        THROW("Creating output tensor for loader failed");
 
-    _internal_images.push_back(output);
+    _internal_tensors.push_back(output);
 
     return output;
 }
 
-Image *
-MasterGraph::create_image(const ImageInfo &info, bool is_output)
-{
-    auto* output = new Image(info);
-    // if the image is not an output image, the image creation is deferred and later it'll be created as a virtual image
-    if(is_output)
-    {
+rocalTensor *
+MasterGraph::create_tensor(const rocalTensorInfo &info, bool is_output) {
+    auto *output = new rocalTensor(info);
+    // if the tensor is not an output tensor, the tensor creation is deferred and later it'll be created as a virtual tensor
+    if(is_output) {
         if (output->create_from_handle(_context) != 0)
-            THROW("Cannot create the image from handle")
+            THROW("Cannot create the tensor from handle")
+        _internal_tensor_list.push_back(output);
```

Comment:
why do we need 2 sets of tensors (internal and output) for each tensor. This will increase the memory requirements twice

Reply:
Hi @rrawther In image pipeline the output is returned to user by getting read buffers from ring buffer and copying the data into user passed buffers.
In the current changes we are maintaining 2 tensor lists in order to directly return the tensor list to the user. 

1. The internal_tensor_list which is used to store a list of the output tensors and used in output routine(handle swapped with the write buffers in ring buffer) . 
2. The output_tensor_list which stores the same list of output tensors and is used to return the tensor list to the user (handle swapped with the read buffers in ring buffer when output list is requested by the user).

Hence we are maintaining two tensor lists for this purpose. We were not able to use a single tensor list, because by the time user requests for the output the internal tensor list is updated with next batch of data